### PR TITLE
Make 'paasta push-to-registry' refuse to overwrite an existing image

### PR DESCRIPTION
**Pull request by oktopuz**
internal ticket: PAASTA-9910

### CODE REVIEW

File: paasta_tools/cli/cmds/push_to_registry.py [hunk 1]
```diff
@@ -83,6 +99,18 @@ def paasta_push_to_registry(args):
...
+        try:
+            if is_docker_image_already_in_registry(service, args.commit):
+                paasta_print("The docker image is already in the PaaSTA docker registry. "
+                             "I'm NOT overriding the existing image. "
+                             "Add --force to override the image in the registry if you are sure what you are doing.")
+                return 0
+        except RequestException as e:
+            paasta_print("Can not connect to the PaaSTA docker registry to verify if this image exists.\n"
+                         "%s" % str(e))
+            return 1
```
Comment (solarkennedy) [inline paasta_tools/cli/cmds/push_to_registry.py]: This is the only thing I would be worried about. We could have a lot broken pipelines if the registry can't talk back in 5 seconds.

Comment (oktopuz) [inline paasta_tools/cli/cmds/push_to_registry.py]: I'm increasing it to 30 seconds.
It is the same timeout as docker uses https://github.com/moby/moby/blob/master/distribution/registry.go#L65

File: paasta_tools/cli/cmds/push_to_registry.py [hunk 2]
```diff
@@ -108,3 +136,43 @@ def paasta_push_to_registry(args):
             level='event',
         )
     return returncode
+
+
+def read_docker_registy_creds(registry_uri):
+    dockercfg_path = os.path.expanduser('~/.dockercfg')
```
Comment (chriskuehl) [inline paasta_tools/cli/cmds/push_to_registry.py]: fwiw new versions of docker use `~/.docker/config.json` instead (though maybe still support this old path)

Comment (oktopuz) [inline paasta_tools/cli/cmds/push_to_registry.py]: The old path is still valid: https://github.com/moby/moby/blob/master/cli/config/config.go#L18

**Pull request merged**